manifest/manifestlist/manifestlist_test.go

```diff
@@ -130,231 +130,77 @@ func TestManifestList(t *testing.T) {
-		t.Fatalf("manifest bytes not equal:\nexpected:\n%s\nactual:\n%s\n", string(expected), string(canonical))
-	}
-
-	// Check that the canonical field has the expected value.
-	if !bytes.Equal([]byte(expectedOCIImageIndexSerialization), canonical) {
-		t.Fatalf("manifest bytes not equal:\nexpected:\n%s\nactual:\n%s\n", expectedOCIImageIndexSerialization, string(canonical))
-	}
-
-	var unmarshalled DeserializedManifestList
-	if err := json.Unmarshal(deserialized.canonical, &unmarshalled); err != nil {
-		t.Fatalf("error unmarshaling manifest: %v", err)
-	}
-
-	if !reflect.DeepEqual(&unmarshalled, deserialized) {
-		t.Fatalf("manifests are different after unmarshaling: %v != %v", unmarshalled, *deserialized)
-	}
-
-	references := deserialized.References()
-	if len(references) != 3 {
-		t.Fatalf("unexpected number of references: %d", len(references))
-	}
-	for i := range references {
-		platform := manifestDescriptors[i].Platform
-		expectedPlatform := &v1.Platform{
-			Architecture: platform.Architecture,
-			OS:           platform.OS,
-			OSFeatures:   platform.OSFeatures,
-			OSVersion:    platform.OSVersion,
-			Variant:      platform.Variant,
-		}
-		if !reflect.DeepEqual(references[i].Platform, expectedPlatform) {
-			t.Fatalf("unexpected value %d returned by References: %v", i, references[i])
-		}
-		references[i].Platform = nil
-		if !reflect.DeepEqual(references[i], manifestDescriptors[i].Descriptor) {
-			t.Fatalf("unexpected value %d returned by References: %v", i, references[i])
-		}
-	}
-}
-
-func mediaTypeTest(t *testing.T, contentType string, mediaType string, shouldError bool) {
-	var m *DeserializedManifestList
-	if contentType == MediaTypeManifestList {
+func mediaTypeTest(contentType string, mediaType string, shouldError bool) func(*testing.T) {
+	return func(t *testing.T) {
+		var m *DeserializedManifestList
 		_, m = makeTestManifestList(t, mediaType)
-	} else {
-		_, m = makeTestOCIImageIndex(t, mediaType)
-	}
 
-	_, canonical, err := m.Payload()
-	if err != nil {
-		t.Fatalf("error getting payload, %v", err)
-	}
-
-	unmarshalled, descriptor, err := distribution.UnmarshalManifest(
-		contentType,
-		canonical)
-
-	if shouldError {
-		if err == nil {
-			t.Fatalf("bad content type should have produced error")
-		}
-	} else {
+		_, canonical, err := m.Payload()
 		if err != nil {
-			t.Fatalf("error unmarshaling manifest, %v", err)
-		}
-
-		asManifest := unmarshalled.(*DeserializedManifestList)
-		if asManifest.MediaType != mediaType {
-			t.Fatalf("Bad media type '%v' as unmarshalled", asManifest.MediaType)
-		}
-
-		if descriptor.MediaType != contentType {
-			t.Fatalf("Bad media type '%v' for descriptor", descriptor.MediaType)
+			t.Fatalf("error getting payload, %v", err)
 		}
 
-		unmarshalledMediaType, _, _ := unmarshalled.Payload()
-		if unmarshalledMediaType != contentType {
-			t.Fatalf("Bad media type '%v' for payload", unmarshalledMediaType)
+		unmarshalled, descriptor, err := distribution.UnmarshalManifest(
```

Comment:
is there a reason why this `func` call spreads to several line instead of being on a single line? Doesn't seem like it's very long.

Reply:
It was before on L295. Happy to fix if you like but at this point I'd prefer keeping the blame view cleaner.